ui/src/components/schemas/SchemaDetails.tsx

```diff
@@ -163,179 +155,100 @@ export function SchemaDetails() {
-                            padding: 16,
-                          }}
-                          language="json"
-                          style={a11yLight}
-                        >
-                          {JSON.stringify(rawJsonSchema, null, 2)}
-                        </SyntaxHighlighter>
-                      );
-                    }
-                  }
-                })()}
+                  <Row justify="space-between">
+                    <Typography.Text type="secondary">Download</Typography.Text>
 
-                <Row justify="end">
-                  <Space size="middle">
                     <Button
-                      icon={<CreditCardIcon />}
                       onClick={() => {
-                        navigate(generatePath(ROUTES.issueCredential.path, { schemaID }));
+                        downloadJsonFromUrl({
+                          fileName: schema.data.name,
+                          url: url,
+                        })
+                          .then(() => {
+                            void message.success("Schema downloaded successfully.");
+                          })
+                          .catch(() => {
+                            void message.error(
+                              "An error occurred while downloading the schema. Please try again."
+                            );
+                          });
                       }}
-                      type="primary"
+                      style={{ height: 24, padding: 0 }}
+                      type="link"
                     >
-                      Issue credential
+                      JSON Schema
                     </Button>
-                  </Space>
-                </Row>
-              </Space>
-            </Card>
+                  </Row>
+                </Space>
+              }
+              jsonLdType={jsonLdType.data}
+              rawJsonLdContext={rawJsonLdContext}
+              rawJsonSchema={rawJsonSchema}
+              schema={schema.data}
+            />
           );
         }
+
+        return (
+          <Card style={{ margin: "auto", maxWidth: CARD_WIDTH }}>
+            {(() => {
+              if (hasAsyncTaskFailed(apiSchema)) {
+                return (
+                  <ErrorResult
+                    error={[
+                      "An error occurred while downloading or parsing the schema from the API:",
+                      apiSchema.error.message,
+                    ].join("\n")}
+                  />
+                );
+              } else if (hasAsyncTaskFailed(schema)) {
+                return <ErrorResult error={schemaErrorToString(schema.error)} />;
+              } else if (hasAsyncTaskFailed(jsonLdType)) {
+                return <ErrorResult error={jsonLdTypeErrorToString(jsonLdType.error)} />;
+              } else if (loading) {
+                return <LoadingResult />;
+              }
+
+              return <ErrorResult error="Unknown error" />;
```

Comment:
I would render everything in the same if/else block as before so that TS can infer that the data is available when the loading has finished with no errors. [Here there's a small PR](https://github.com/0xPolygonID/sh-id-platform/pull/264/files) changing this at the only cost of repeating the `Card`'s JSX.

I've also removed the separate states for the `rawJsonSchema` and the `rawJsonLdContext` and added them into the async tasks' result so they can't be `undefined` when the async tasks are successfully finished.

I've also renamed `getJsonLdTypesFromUrl` as `getSchemaJsonLdTypes` and changed its signature to remove the URL param which can be obtained from the Schema.

Reply:
That's fine, with a small [objection](https://github.com/0xPolygonID/sh-id-platform/pull/264#discussion_r1150283389).